cmd/installer/cli/install_config.go

```diff
@@ -0,0 +1,222 @@
+package cli
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"os"
+
+	"github.com/google/uuid"
+	"github.com/replicatedhq/embedded-cluster/cmd/installer/goods"
+	newconfig "github.com/replicatedhq/embedded-cluster/pkg-new/config"
+	"github.com/replicatedhq/embedded-cluster/pkg-new/tlsutils"
+	"github.com/replicatedhq/embedded-cluster/pkg/airgap"
+	"github.com/replicatedhq/embedded-cluster/pkg/configutils"
+	"github.com/replicatedhq/embedded-cluster/pkg/helpers"
+	"github.com/replicatedhq/embedded-cluster/pkg/netutils"
+	"github.com/replicatedhq/embedded-cluster/pkg/prompts"
+	"github.com/replicatedhq/embedded-cluster/pkg/runtimeconfig"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+// Hop: buildInstallFlags maps cobra command flags to install flags
+func buildInstallFlags(cmd *cobra.Command, flags *installFlags) error {
+	// Target defaulting (if not V3)
+	if !isV3Enabled() {
+		flags.target = "linux"
+	}
+
+	// Target validation
+	if flags.target != "linux" && flags.target != "kubernetes" {
+		return fmt.Errorf(`invalid --target (must be one of: "linux", "kubernetes")`)
+	}
+
+	// If only one of cert or key is provided, return an error
+	if (flags.tlsCertFile != "" && flags.tlsKeyFile == "") || (flags.tlsCertFile == "" && flags.tlsKeyFile != "") {
+		return fmt.Errorf("both --tls-cert and --tls-key must be provided together")
+	}
+
+	// Skip host preflights from env var (if flag not explicitly set)
+	if !cmd.Flags().Changed("skip-host-preflights") {
+		if os.Getenv("SKIP_HOST_PREFLIGHTS") == "1" || os.Getenv("SKIP_HOST_PREFLIGHTS") == "true" {
+			flags.skipHostPreflights = true
+		}
+	}
+
+	// Network interface auto-detection (if not provided)
+	if flags.networkInterface == "" && flags.target == "linux" {
+		autoInterface, err := newconfig.DetermineBestNetworkInterface()
+		if err == nil {
+			flags.networkInterface = autoInterface
+		}
+		// If error, leave empty and validation will catch it later
+	}
+
+	// Port conflict validations
+	if flags.managerPort != 0 && flags.adminConsolePort != 0 {
+		if flags.managerPort == flags.adminConsolePort {
+			return fmt.Errorf("manager port cannot be the same as admin console port")
+		}
+	}
+
+	if flags.localArtifactMirrorPort != 0 && flags.adminConsolePort != 0 {
+		if flags.localArtifactMirrorPort == flags.adminConsolePort {
+			return fmt.Errorf("local artifact mirror port cannot be the same as admin console port")
+		}
+	}
+
+	// CIDR configuration
+	cidrCfg, err := cidrConfigFromCmd(cmd)
+	if err != nil {
+		return err
+	}
+	flags.cidrConfig = cidrCfg
```

Comment:
shouldnt this move to the install config? it is derived/computed

Reply:
These come from flags but also computed and i didn't want to duplicate fields in both structs to avoid confusion when each should be used